python-spec/src/somacore/coordinates.py

```diff
@@ -0,0 +1,381 @@
+"""Definitions of types related to coordinate systems."""
+
+import abc
+import collections.abc
+from typing import Optional, Sequence, Tuple, Union
+
+import attrs
+import numpy as np
+import numpy.typing as npt
+
+from .types import str_or_seq_length
+from .types import to_string_tuple
+
+
+@attrs.define(frozen=True)
+class Axis:
+    """A description of an axis of a coordinate system
+
+    Lifecycle: experimental
+    """
+
+    name: str
+    """Name of the axis."""
+    unit: Optional[str] = None
+    """Optional string name for the units of the axis."""
+
+
+@attrs.define(frozen=True)
+class CoordinateSpace(
```

Comment:
Can we also implement `__eq__` for `CoordinateSpace` in here?

Reply:
coming back from the dead to say that attrs gives you `__eq__` for free